enums.rb

```diff
@@ -0,0 +1,128 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # rubocop:enable Metrics/ModuleLength
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_all?(arg = nil)
+    if block_given? && !arg.nil?
+      return false unless my_all?(arg) && my_all?
+    end
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+    elsif arg.nil?
+      my_each { |i| return false unless i }
+    elsif arg.class == Class
+      my_each { |i| return false unless i.class <= arg }
+    elsif arg.class == Regexp
+      my_each { |i| return false unless i =~ arg }
+    else
+      my_each { |i| return false unless i == arg && i.class <= arg.class }
+    end
+
+    true
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |i|
+      new_array.push(i) if yield(i)
+    end
+    new_array
+  end
+
+  def my_any?(arg = nil)
+    if block_given? && !arg.nil?
+      return true unless my_any?(arg) && my_any?
+    end
+
+    if block_given?
+      my_each { |i| return true unless yield(i) }
+    elsif arg.nil?
+      my_each { |i| return true unless i }
+    elsif arg.class == Class
+      my_each { |i| return true unless i.class <= arg }
+    elsif arg.class == Regexp
+      my_each { |i| return true unless i =~ arg }
+    else
+      my_each { |i| return true unless i == arg && i.class <= arg.class }
+    end
+
+    false
+  end
+
+  def my_none?(arg = nil)
+    my_each do |i|
+      if block_given?
+        return false if yield(i)
+      elsif arg.class == Regexp
+        my_each { return true unless i =~ arg }
+      elsif !i.nil? && i != false
+      else
+        return false
+      end
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    each do |i|
+      if block_given?
+        count += 1 if yield(i) == true
+      elsif !arg.nil?
+        count += 1 if i == arg
+      else
+        count += 1
+      end
+    end
+    count
+  end
+
+  def my_map(&arg)
+    return to_enum(:my_map) unless block_given?
+
+    modified_array = []
+    my_each { |i| modified_array << arg.call(i) }
+
+    modified_array
+  end
+
+  def my_inject(*arg)
+    # rubocop:enable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    raise LocalJumpError unless block_given? || !arg.empty?
+
+    sym = arg.pop unless block_given?
+    my_array = arg + to_a
+    memo = my_array.shift
+    if block_given?
+      my_array.my_each { |el| memo = yield(memo, el) }
+    else
+      my_array.my_each { |el| memo = memo.send(sym, el) }
+    end
+    memo
+  end
+
+  def multiply_els
+    my_inject { |i, a| i * a }
+  end
```

Comment:
- [ ] should have working `multiply_els` that uses `my_inject` as described on Odin ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/multiply_els.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### multiply_els

- [ ] multiply_els accepts an array as an argument and multiplies all the elements of the array together using #my_inject

- Your multiple_els it's a method that takes an argument and retrieves a number is not an enumerable method so you should place it outside the enumerable module and it should look something like this:

```ruby
def mutiply_els(arg)
  result = #your code that does something with arg here 
end
```